File: app/src/main/java/edu/ksu/wheatgenetics/verify/ScanActivity.java
package edu.ksu.wheatgenetics.verify;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.KeyEvent;
import android.view.MenuItem;
import android.widget.ImageView;

import com.google.zxing.ResultPoint;

import com.journeyapps.barcodescanner.BarcodeCallback;
import com.journeyapps.barcodescanner.BarcodeResult;
import com.journeyapps.barcodescanner.DecoratedBarcodeView;

import java.util.List;

public class ScanActivity extends AppCompatActivity {

    private DecoratedBarcodeView barcodeScannerView;
    private String lastText;

    private BarcodeCallback callback = new BarcodeCallback() {

        @Override
        public void barcodeResult(BarcodeResult result) {

            if (result.getText() == null || result.getText().equals(lastText)) return;

            lastText = result.getText();
            barcodeScannerView.setStatusText(result.getText());

            ImageView imageView = (ImageView) findViewById(R.id.barcodePreview);
            imageView.setImageBitmap(result.getBitmapWithResultPoints(Color.GREEN));

            final Intent i = new Intent();
            i.putExtra(VerifyConstants.CAMERA_RETURN_ID, result.getText());

            setResult(RESULT_OK, i);
            finish();

        }

        @Override
        public void possibleResultPoints(List<ResultPoint> resultPoints) {

        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_capture);
        barcodeScannerView = (DecoratedBarcodeView)
                findViewById(R.id.zxing_barcode_scanner);
        barcodeScannerView.getBarcodeView().getCameraSettings().setContinuousFocusEnabled(true);
        barcodeScannerView.getBarcodeView().getCameraSettings().setBarcodeSceneModeEnabled(true);
        barcodeScannerView.decodeContinuous(callback);

        if(getSupportActionBar() != null){
            getSupportActionBar().setTitle(null);
            getSupportActionBar().getThemedContext();
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setHomeButtonEnabled(true);
        }
    }


    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        switch (item.getItemId()) {
            case android.R.id.home:
                setResult(RESULT_OK);
                finish();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    @Override
    protected void onResume() {
        super.onResume();

        barcodeScannerView.resume();
    }

    @Override
    protected void onPause() {
        super.onPause();

        barcodeScannerView.pause();
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {

        return barcodeScannerView.onKeyDown(keyCode, event) || super.onKeyDown(keyCode, event);
    }
}

File: app/src/main/java/org/phenoapps/intercross/ScanActivity.kt
package org.phenoapps.intercross

import android.app.Activity
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.KeyEvent
import android.view.MenuItem
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity

import com.google.zxing.ResultPoint

import com.journeyapps.barcodescanner.BarcodeCallback
import com.journeyapps.barcodescanner.BarcodeResult
import com.journeyapps.barcodescanner.DecoratedBarcodeView
import org.phenoapps.intercross.IntercrossActivity.Companion.CAMERA_RETURN_ID

class ScanActivity : AppCompatActivity() {

    private var barcodeScannerView: DecoratedBarcodeView? = null
    private var lastText: String? = null

    private val callback = object : BarcodeCallback {

        override fun barcodeResult(result: BarcodeResult) {

            if (result.text == null || result.text == lastText) return

            lastText = result.text
            barcodeScannerView!!.setStatusText(result.text)

            val imageView = findViewById(org.phenoapps.intercross.R.id.barcodePreview) as ImageView
            imageView.setImageBitmap(result.getBitmapWithResultPoints(Color.GREEN))

            val i = Intent()
            i.putExtra(CAMERA_RETURN_ID, result.text)

            setResult(Activity.RESULT_OK, i)
            finish()

        }

        override fun possibleResultPoints(resultPoints: List<ResultPoint>) {

        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)

        setContentView(org.phenoapps.intercross.R.layout.activity_capture)
        barcodeScannerView = findViewById(org.phenoapps.intercross.R.id.zxing_barcode_scanner) as DecoratedBarcodeView
        barcodeScannerView!!.barcodeView.cameraSettings.isContinuousFocusEnabled = true
        barcodeScannerView!!.barcodeView.cameraSettings.isBarcodeSceneModeEnabled = true
        barcodeScannerView!!.decodeContinuous(callback)

        if (supportActionBar != null) {
            supportActionBar!!.title = null
            supportActionBar!!.themedContext
            supportActionBar!!.setDisplayHomeAsUpEnabled(true)
            supportActionBar!!.setHomeButtonEnabled(true)
        }
    }


    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        when (item.itemId) {
            android.R.id.home -> {
                setResult(Activity.RESULT_OK)
                finish()
                return true
            }
            else -> return super.onOptionsItemSelected(item)
        }
    }

    override fun onResume() {
        super.onResume()

        barcodeScannerView!!.resume()
    }

    override fun onPause() {
        super.onPause()

        barcodeScannerView!!.pause()
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {

        return barcodeScannerView!!.onKeyDown(keyCode, event) || super.onKeyDown(keyCode, event)
    }
}